File: src/main/java/com/cfgglobal/test/service/rule/access/AdminAccessRule.java
package com.cfgglobal.test.service.rule.access;


import com.cfgglobal.test.base.Filter;
import com.cfgglobal.test.domain.Permission;
import org.springframework.stereotype.Component;

@Component
public class AdminAccessRule extends AbstractAccessRule {
    @Override
    public Filter exec(Permission permission) {
        return Filter.EMPTY;
    }

    @Override
    public String getRuleName() {
        return "admin";
    }
}

File: src/main/kotlin/com/cfgglobal/test/service/rule/access/AdminAccessRule.kt
package com.cfgglobal.test.service.rule.access


import com.cfgglobal.test.domain.Permission
import com.cfgglobal.test.vo.Filter
import org.springframework.stereotype.Component

@Component
class AdminAccessRule : AbstractAccessRule() {

    override val ruleName: String
        get() = "admin"

    override fun exec(permission: Permission): Filter {
        return Filter.EMPTY
    }
}
